okcupyd/filter.py

```diff
@@ -179,30 +179,33 @@ def my_filter_function(argument):
                 return '4,{0}'.format(argument)
 
         :param function: The filter function to register.
+        :param output_key: The key to use to output the provided value. Will
+                           default to the only value in keys if keys has
+                           length 1.
         :param keys: Keys that should be used as the argument names for
                      `function`, if none are provided, the filter functions
                      argument names will be used instead.
+        :type keys: iterable of strings
         :param decider: a function of signature
                         `(function, incoming_keys, accepted_keys)` that returns
                         True if the filter function should be called and False
                         otherwise. Defaults to
                         :meth:`~.all_not_none_decider`
-        :param acceptable_values: A list of acceptable values for the parameter
-                                  of the filter function (or a list of lists if
-                                  the filter function takes multiple parameters)
-        :param types: The type of the parameter accepted by the incoming filter
-                      function (or a list of types if the function takes
-                      multiple parameters)
         :param descriptions: A description for the incoming filter function's
                              argument (or a list of descriptions if the filter
                              function takes multiple arguments)
-        :param output_key: The key to use to output the provided value. Will
-                           default to the only value in keys if keys has
-                           length 1.
+        :param acceptable_values: An iterable of acceptable values for the parameter
+                                  of the filter function (or a list of iterables if
+                                  the filter function takes multiple parameters).
+                                  Docstring will use the repr of the iterable's elements.
+        :param types: The type (or a string describing the type) of the parameter accepted
+                      by the incoming filter function (or a list of types if the function
+                      takes multiple parameters)
         """
         kwargs['transform'] = function
-        if kwargs.get('decider'):
-            kwargs['decide'] = kwargs.get('decider')
+        # assert 'decide' not in kwargs
```

Comment:
You're looking at only part of the diff. One thing I did was change the test condition to `if 'decider' in kwargs`. The other thing I did was add the (inactive) comment, to catch a case where a programmer was confused about the interface and provided both `decide=` and `decider=` kwargs. In my version the relevant lines end up being:

    # assert 'decide' not in kwargs
    if 'decider' in kwargs:
        kwargs['decide'] = kwargs.pop('decider')